clash-lib/src/Clash/Core/Util.hs

```diff
@@ -982,11 +983,24 @@ shouldSplit0 tcm (TyConApp tcNm tyArgs)
   , [dc] <- tyConDataCons tc
   , let dcArgs  = substArgTys dc tyArgs
   , let dcArgVs = map (tyView . coreView tcm) dcArgs
-  = if any (\ty -> isJust (shouldSplit0 tcm ty) || splitTy ty) dcArgVs then
+  = if any shouldSplitTy dcArgVs && not (isHidden tcNm tyArgs) then
```

Comment:
Could you confirm that this does the right thing for:

```haskell
type MyMischievousHidden dom = 
  (  ( Hidden (AppendSymbol dom "_clk" (Clock dom))
     , Hidden (AppendSymbol dom "_rst") (Reset dom) )
  , KnowDomain dom
  )
```

I think it does do the right thing, but I'd like some confirmation. 

However, it wont work for a use who does this I think:

```haskell
type MyMischievousHiddenClk dom
  = ( KnownDomain dom
    , Hidden (AppendSymbol dom "_clk") (Clock dom)
    )

Reply:
Correct, it won't handle the latter case. (And handling the first case is a happy accident.) It was specifically made to recognize `HiddenClock` & co. I'm happy to make it more general, but I'm wondering whether it's worth the extra engineering time.

Reply:
I can live with a `-- TODO` comment, and a new issue on the issue tracker.

Reply:
Ah, it doesn't handle the first case either. It's specifically `(ClockOrResetOrEnable, KnownDomain)`.

Reply:
Done. (Will add the issue after merging.)